### DISCUSSION TITLE
Data source for Chart

### DISCUSSION THREAD

**Discussion by sladerose** [Budibase/budibase#1614] [Help] [answered]
I'm trying to populate a chart with internal data, but can't seem to figure out how to. Could anyone give me a hand? When I click "provider" it show nothing, even though I have two internal tables. Thank you in advance! 
<img width="1397" alt="Screenshot 2021-06-02 at 19 10 24" src="https://user-images.githubusercontent.com/72910898/120523467-33898400-c3d6-11eb-9f12-f87723b86ef7.png">

**Comment by joebudi** [comment, marked as answer]
Could you try:
- adding a data provider component
- bind data provider to your internal table
- add a chart component
- bind chart component to your data provider

**Reply by sladerose** [reply]
Thank you very much. I tried that and figured out that I needed to make the chart a child of the data provider. :)

**Reply by joebudi** [reply]
That's great. Happy building!

**Reply by sladerose** [reply]
Thanks Joe! Our team is absolutely loving Budibase, we still can't believe
it's open source...

I had another question. We're running Budibase on Docker desktop on our
Macbooks, but when we try to update Budibase, we're getting this error:

```
Error - Please run the hosting --init command before any other hosting
command.
```

If you could point us in the right direction so that we can receive all the
new and amazing updates, we'd greatly appreciate it. :)

**Reply by mike12345567** [reply]
Hi @sladerose I assume you're trying to run the command `budi hosting --update`, for this command to work it needs to be ran in the same directory where you downloaded your `docker-compose.yml` or first ran `budi hosting --init`.
